packages/react/src/components/Notification/Notification.tsx

```diff
@@ -50,19 +59,42 @@ function useEscapeToClose(ref, callback, override = true) {
   });
 }
 
+export interface NotificationActionButtonProps {
+  /**
+   * Specify the content of the notification action button.
+   */
+  children?: ReactNode;
+
+  /**
+   * Specify an optional className to be applied to the notification action button
+   */
+  className?: string;
+
+  /**
+   * Specify if the visual treatment of the button should be for an inline notification
+   */
+  inline?: boolean;
+
+  /**
+   * Optionally specify a click handler for the notification action button.
+   */
+  onClick?(): void;
+}
+
```

Comment:
This should also allow for `Button` props through `...rest`, should this PR be blocked until Button has been migrated into typescript? 🤔
CC @tay1orjones